src/ui/public/sortable_column/sortable_column.js

```diff
@@ -0,0 +1,38 @@
+import { uiModules } from 'ui/modules';
+import template from './sortable_column.html';
+import './sortable_column.less';
+
+const app = uiModules.get('kibana');
+
+app.directive('sortableColumn', function () {
+  return {
+    restrict: 'E',
+    transclude: true,
+    template: template,
+    scope: {
+      field: '@',
+      sortField: '=',
+      sortReverse: '=',
+      onSortChange: '=',
+    },
+    controllerAs: 'sortableColumn',
+    bindToController: true,
+    controller: class SortableColumnController {
+      toggle = () => {
+        if (this.sortField === this.field) {
+          this.onSortChange(this.field, !this.sortReverse);
+        } else {
+          this.onSortChange(this.field, false);
+        }
+      }
+
+      get isSortedAscending() {
+        return (this.sortField === this.field) && (!this.sortReverse);
+      }
+
+      get isSortedDescending() {
```

Comment:
How do you feel about making these regular scope methods instead of getters? I mentioned something about this in another PR.